stdlib/public/runtime/SwiftObject.mm

```diff
@@ -1352,6 +1351,32 @@ static bool usesNativeSwiftReferenceCounting_nonNull(
     swift_isUniquelyReferenced_nonNull_native((const HeapObject*)object);
 }
 
+#if SWIFT_OBJC_INTEROP
+// It would be nice to weak link instead of doing this, but we can't do that
+// until the new API is in the versions of libobjc that we're linking against.
+static bool isUniquelyReferenced(const void *object) {
+  static auto objcIsUniquelyRefd = reinterpret_cast<bool (*)(const void *)>(
+      dlsym(RTLD_DEFAULT, "objc_isUniquelyReferenced"));
```

Comment:
This should use `RTLD_NEXT` to limit what `dlsym` has to search.

The codegen for local static initializers isn't the best. This should use `SWIFT_LAZY_CONSTANT` which will use `dispatch_once`. Or there's always my favorite trick of setting the initial value of the function pointer to a function that does the lookup and sets the real value, like with `dispatchEnqueueFuncType`. https://github.com/apple/swift/blob/main/stdlib/public/Concurrency/GlobalExecutor.cpp#L214.

It might be worth adding a define to the header like `OBJC_ISUNIQUELYREFERENCED_DEFINED` and then use that to directly reference the symbol when it's available. That makes it less urgent to remove the `dlsym`. We're doing this in a couple of other places already, look for `OBJC.*DEFINED`. Incidentally, some of those can probably be removed if you're bored and looking for something quick to do.

Reply:
I used `SWIFT_LAZY_CONSTANT`. I've added that macro as well.